src/receive_imf.rs

```diff
@@ -1564,7 +1565,7 @@ INSERT INTO msgs
 ON CONFLICT (id) DO UPDATE
 SET rfc724_mid=excluded.rfc724_mid, chat_id=excluded.chat_id,
     from_id=excluded.from_id, to_id=excluded.to_id, timestamp_sent=excluded.timestamp_sent,
-    type=excluded.type, state=min(state,max(?,13)), msgrmsg=excluded.msgrmsg,
+    type=excluded.type, state=min(state,excluded.state), msgrmsg=excluded.msgrmsg,
```

Comment:
As for this, i agree that `excuded.state` should be used instead of `?`, but i prefer to have `max(...,13)` here because we never want to downgrade the state to `InFresh`. So this is a bit cumbersome, but more straightforward